tests/test_archive.py

```diff
@@ -0,0 +1,145 @@
+import os
+
+from sqlalchemy import create_engine
+
+from library import recipe_engine
+from library.archive import Archive
+
+from . import test_root_path
+
+a = Archive()
+pg = create_engine(recipe_engine)
+
+
+def start_clean(local_files: list, s3_files: list):
+    for f in local_files:
+        if os.path.isfile(f):
+            os.remove(f)
+    for f in s3_files:
+        if a.s3.exists(f):
+            a.s3.rm(f)
+
+
+def test_archive_1():
+    local_not_exist = [
+        ".library/datasets/nypl_libraries/20210122/nypl_libraries.csv.zip",
+        ".library/datasets/nypl_libraries/20210122/nypl_libraries.csv",
+    ]
+    s3_exist = [
+        "datasets/nypl_libraries/20210122/nypl_libraries.csv.zip",
+        "datasets/nypl_libraries/latest/nypl_libraries.csv.zip",
+        "datasets/nypl_libraries/20210122/config.yml",
+        "datasets/nypl_libraries/20210122/config.json",
+        "datasets/nypl_libraries/latest/nypl_libraries.csv.zip",
+        "datasets/nypl_libraries/latest/config.yml",
+        "datasets/nypl_libraries/latest/config.json",
+    ]
+    start_clean(local_not_exist, s3_exist)
+    a(
+        f"{test_root_path}/data/nypl_libraries.yml",
+        output_format="csv",
+        push=True,
+        clean=True,
+        latest=True,
+        compress=True,
+    )
+    for f in local_not_exist:
+        assert not os.path.isfile(f)
+    for f in s3_exist:
+        assert a.s3.exists(f)
+    start_clean(local_not_exist, s3_exist)
+
+
+def test_archive_2():
+    s3_not_exist = [
+        "datasets/nypl_libraries/20210122/nypl_libraries.geojson.zip",
+        "datasets/nypl_libraries/20210122/config.yml",
+        "datasets/nypl_libraries/20210122/config.json",
+        "datasets/nypl_libraries/20210122/nypl_libraries.geojson",
+        "datasets/nypl_libraries/latest/nypl_libraries.geojson.zip",
+        "datasets/nypl_libraries/latest/config.yml",
+        "datasets/nypl_libraries/latest/config.json",
+        "datasets/nypl_libraries/latest/nypl_libraries.geojson",
+    ]
+    local_exist = [
+        ".library/datasets/nypl_libraries/20210122/nypl_libraries.geojson.zip",
```

Comment:
Should the non-zipped geojson also exist locally in this check? This call is using clean=False, and by default the geojson importer using inplace=False.

Reply:
you are right

Reply:
we are passing these variables back and forth across different classes, and it's easy to lose track. Let me know if you have ideas about how to set it up so it's straight forward